### PR TITLE
chore: simplify next share index

### PR DESCRIPTION
**Pull request by cmwaters**
This PR pulls out a lot of redundant logic, simplifying the `NextShareIndex` function to make it more readable and understandable. I've also added more comments and updated the links.

Rationale: There is no subtreeWidth value that is not a multiple of the square size given that both must be powers of 2 and that the subtreewidth can't be greater than the minimum blob square size (and the blob has to fit in the square). This reinforces the notion that this algorithm is square size independent

### CODE REVIEW

**Review by evan-forbes — APPROVED** [review]
good find and makes sense that we can refactor this. iirc it was originally written for the first version of non-interactive defaults

File: pkg/shares/blob_share_commitment_rules.go [hunk 1]
```diff
@@ -45,35 +45,31 @@ func BlobSharesUsedNonInteractiveDefaults(cursor, squareSize, subtreeRootThresho
...
 //
-// https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md#non-interactive-default-rules
+// See https://github.com/celestiaorg/celestia-app/blob/main/specs/src/specs/data_square_layout.md
+// for more information.
 func NextShareIndex(cursor, blobShareLen, squareSize, subtreeRootThreshold int) int {
 	// if we're starting at the beginning of the row, then return as there are
 	// no cases where we don't start at 0.
 	if isStartOfRow(cursor, squareSize) {
 		return cursor
 	}
```
Comment (rootulp) [inline pkg/shares/blob_share_commitment_rules.go:59]: [optional] this optimization doesn't seem necessary. All tests pass with it's removal and the specs don't describe it.
```suggestion
```

**Review by rootulp — APPROVED** [review]
Nice clean up!

File: pkg/shares/blob_share_commitment_rules.go (hunk 1 above)
Comment (cmwaters) [reply pkg/shares/blob_share_commitment_rules.go:59]: Yeah it's a minor optimisation. I don't mind removing it

**Review by evan-forbes — APPROVED** [review]

**Review by rootulp — APPROVED** [review]

**Pull request merged**